src/content/contribute/writing-a-scaffold.md

```diff
@@ -0,0 +1,400 @@
+
+
+## Some more configs
+
+Let's start by looking at `dev-config.js`. We have some answers, now we want to use them to build up a config. How do we do that? Evidently, we should mount our values on the variable we've declared, with some properties we want to build up. We also want to use the answers for the entry prop. I've also taken the liberty to add an output property that has a `filename`.
+
+T> With string values, you need to wrap your strings once again. This is because we can declare some other functionality, using only " ", while " 'Mystring' " resolves to a string.
+
+__dev-config.js__
+
+```js
+module.exports = function createDevConfig(answer) {
+  let entryProp = answer.entry ? ( '\'' + answer.entry + '\'') : '\'index.js\'';
+  let devConfig = {
+    entry: entryProp,
+    output: {
+      filename: '\'[name].js\''
+    }
+  };
+  return devConfig;
+};
+
+```
+
+Try running `webpack init webpack-scaffold-demo`, and now you're seeing your first scaffold!
+
+
+
+## Basic Scaffold
+
+Cool. Now we've got an initial scaffold. Let's add the rest of our options! For the context, let's say we've got a `path.join` we want to make use of. For this, you use a single quote string.
+
+```js
+module.exports = function createDevConfig(answer) {
+  let entryProp = answer.entry ? ( '\'' + answer.entry + '\'') : '\'index.js\'';
+  let devConfig = {
+    entry: entryProp,
+    output: {
+      filename: '\'[name].js\''
+    },
+    context: 'path.join(__dirname, "src")'
+  };
+  return devConfig;
+};
+```
+
+
+
+## Add more functionality
+
+We're ready to add a plugin, as the last piece of the puzzle. For this, I'm going to create a utility for commonsChunk, based on the input from the user. I'll start by adding another question to our prompt.
+
+```js
+const Generator = require('yeoman-generator');
+const List = require('@webpack-cli/webpack-scaffold').List;
+const Input = require('@webpack-cli/webpack-scaffold').Input;
+const createDevConfig = require('./dev-config');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {
+        webpackOptions: {}
+      }
+    };
+  }
+
+  prompting() {
+    return this.prompt([
+      List('confirm', 'Welcome to the demo scaffold! Are you ready?', ['Yes', 'No', 'Pengwings']),
+      Input('entry', 'What is the entry point in your app?'),
+      Input('plugin', 'What do you want to name your commonsChunk?')
+    ]).then (answer => {
+      if(answer['confirm'] === 'Pengwings') {
+        this.options.env.configuration.dev.webpackOptions = createDevConfig(answer);
+      }
+    });
+  }
+};
+```
+
+
+
+## Create string with chunks
+
+Now, we've got to create a string with our chunk. This is how it looks.
+
+```js
+module.exports = function createCommonsChunkPlugin(chunk) {
+  return (
+    'new webpack.optimize.CommonsChunkPlugin({name:' + '\'' + chunk + '\'' +
```

Comment:
This needs change, commonsChunkPlugin is deprecared in favor of SplitChunksPlugin